src/es-router.js

```diff
@@ -16,11 +17,13 @@ class EsRouter {
     this.useHash = useHash;
     this.routes = routes;
     this.base = base;
-    this.notStrictRouting = notStrictRouting;
+    this.strictRouting = strictRouting;
     this.queryParams = this.getParamsFromUrl();
 
-    if (base && base[base.length - 1] === '/') {
+    if (base && base[base.length - 1] === '/' && base !== '/') {
```

Comment:
won't the second case take care of the third? if the default of `base` is `/` then the length would be one, with a minus one would return `/`